app/vlstorage/netselect/netselect.go

```diff
@@ -313,6 +318,11 @@ func (s *Storage) MustStop() {
 	s.sns = nil
 }
 
+// WriteMetrics writes internal storage metrics to the provided writer in Prometheus text exposition format.
+func (s *Storage) WriteMetrics(w io.Writer) {
+	metrics.WriteGaugeUint64(w, `vl_select_remote_send_errors_total`, s.remoteSendErrors.Load())
```

Comment:
It would be great to have per-storage-node counters on the number of `send` errors. This can be done by calling `metrics.GetOrCreateCounter()` function at the corresponding per-storage-node goroutine inside `Storage.runQuery()`.

The same applies to `vl_insert_remote_send_errors_total` counter.

This will also simplify the code a bit.